src/main/java/ezvcard/io/json/JCardWriter.java

```diff
@@ -1,224 +1,286 @@
+	}
+
+	/**
+	 * @param file the file to write to (UTF-8 encoding will be used)
+	 * @param wrapInArray true to enclose all written vCards in a JSON array,
+	 * false not to
+	 * @throws IOException if there's a problem opening the file
+	 */
+	public JCardWriter(File file, boolean wrapInArray) throws IOException {
+		this(utf8Writer(file), wrapInArray);
+	}
+
+	/**
+	 * @param writer the writer to write to
+	 */
+	public JCardWriter(Writer writer) {
+		this(writer, false);
+	}
+
+	/**
+	 * @param writer the writer to write to
+	 * @param wrapInArray true to enclose all written vCards in a JSON array,
+	 * false not to
+	 */
+	public JCardWriter(Writer writer, boolean wrapInArray) {
+		this.writer = new JCardRawWriter(writer, wrapInArray);
+	}
+
+	/**
+	 * @param generator the generator to write to
+	 */
+	public JCardWriter(JsonGenerator generator) {
+		this.generator  = generator;
+		this.writer = new JCardRawWriter(generator);
+	}
+
+	/**
+	 * Writes a vCard to the stream.
+	 * @param vcard the vCard that is being written
+	 * @param properties the properties to write
+	 * @throws IOException if there's a problem writing to the output stream
+	 * @throws IllegalArgumentException if a scribe hasn't been registered for a
+	 * custom property class (see: {@link #registerScribe registerScribe})
+	 */
+	@Override
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	protected void _write(VCard vcard, List<VCardProperty> properties) throws IOException {
+
+		Object previousValue = getCurrentValue();
+		setCurrentValue(vcard);
+
+		writer.writeStartVCard();
+		setCurrentValue(targetVersion);
+		writer.writeProperty("version", VCardDataType.TEXT, JCardValue.single(targetVersion.getVersion()));
+		setCurrentValue(vcard);
+
+		for (VCardProperty property : properties) {
+			VCardPropertyScribe scribe = index.getPropertyScribe(property);
+
+			//marshal the value
+			JCardValue value;
+			try {
+				value = scribe.writeJson(property);
+			} catch (SkipMeException e) {
+				//property has requested not to be written
+				continue;
+			} catch (EmbeddedVCardException e) {
+				//don't write because jCard does not support embedded vCards
+				continue;
+			}
+
+			String group = property.getGroup();
+			String name = scribe.getPropertyName().toLowerCase();
+			VCardParameters parameters = scribe.prepareParameters(property, targetVersion, vcard);
+			VCardDataType dataType = scribe.dataType(property, targetVersion);
+
+			setCurrentValue(property);
+			writer.writeProperty(group, name, parameters, dataType, value);
+			setCurrentValue(vcard);
+		}
+
+		writer.writeEndVCard();
+
+		setCurrentValue(previousValue);
+	}
+
+	/**
+	 * If this object has a {@link JsonGenerator), and the generator has an
+	 * output context, gets the current value of the output context.
+	 * 
+	 * @return the value of the object that is currently being serialized, if
+	 *         available
+	 */
+	private Object getCurrentValue() {
+		if (generator != null) {

```

Comment:
A ternary operator would be more concise here:

`return (generator == null) ? null  : generator.getCurrentValue();`